tests/timeseries/_battery_pool/test_battery_pool.py

```diff
@@ -438,7 +438,11 @@ async def run_test_battery_status_channel(  # pylint: disable=too-many-arguments
     msg = await asyncio.wait_for(
         battery_pool_metric_receiver.receive(), timeout=waiting_time_sec
     )
-    assert msg is None
+    if isinstance(msg, PowerMetrics):
+        assert msg.inclusion_bounds is None
+        assert msg.exclusion_bounds is None
+    elif isinstance(msg, Sample):
+        assert msg.value is None
```

Comment:
does just saying `else:` also work here?  you can test locally with `nox -R -e mypy`.

If not,  we'll probably have to add a separate `else:` that always raises,  so that future changes to `T` don't pass through the gaps.

Reply:
> does just saying else: also work here? you can test locally with nox -R -e mypy.

It does not, because then it does not know what the type of `msg` is, and whether or not it has the `value` field. I will add the additional `else` statement.